src/sentry/integrations/vercel/webhook.py

```diff
@@ -44,4 +103,104 @@ def post(self, request):
             logger.error("vercel.webhook.invalid-signature")
             return self.respond(status=401)
 
-        return self.respond(status=200)
+        data = request.data
+        payload = data["payload"]
+        external_id = data.get("teamId") or data["userId"]
+        vercel_project_id = payload["projectId"]
+
+        logging_params = {"external_id": external_id, "vercel_project_id": vercel_project_id}
+
+        if payload["target"] != "production":
+            logger.info(
+                "Ignoring deployment for environment: %s" % payload["target"], extra=logging_params
+            )
+            return self.respond(status=202)
+
+        # Steps:
+        # 1. find all org integrations that match the external id
+        # 2. search the configs to find one that matches the vercel project of the webhook
+        # 3. Look up the Sentry project that matches
+        # 4. Look up the connected internal integration
+        # 5. find the token associated with that installation
+        # 6. Determine the commit sha and repo based on what provider is used
+        # 7. Create the release using the token WITHOUT refs
+        # 8. Update the release with refs
+
+        # find all org integrations that match the external id
+        try:
+            org_integrations = OrganizationIntegration.objects.select_related(
+                "organization"
+            ).filter(integration__external_id=external_id, integration__provider=self.provider)
+        except OrganizationIntegration.DoesNotExist:
+            logger.info("Integration not found", extra=logging_params)
+            return self.respond({"detail": "Integration not found"}, status=404)
+
+        # for each org integration, search the configs to find one that matches the vercel project of the webhook
+        for org_integration in org_integrations:
+            project_mappings = org_integration.config.get("project_mappings") or []
```

Comment:
if there are no project mappings set up should be responding with an error instead of a 202?

Reply:
@MeredithAnya This is a totally valid use case if someone deploys before setting up the project mappings

Reply:
@scefali true, I am wondering about the case whether they don't finish setting it up - is that possible that people could unknowingly not have the mappings set up. maybe that's an unlikely scenario

Reply:
@MeredithAnya Yea, not having the mappings is definitely a potential problem. I think the solution to that is putting this information in a dashboard that we show in the UI hook. Either way, returning a `404` on the webhook is not going to let the user know they need to take action.